lib/Dialect/Pipeline/Transforms/ExplicitRegs.cpp

```diff
@@ -36,25 +36,20 @@ class ExplicitRegsPass : public ExplicitRegsBase<ExplicitRegsPass> {
   // operations containing the requested regs, and the backedge will be
   // replaced. MapVector ensures deterministic iteration order, which in turn
   // ensures determinism during stage op IR emission.
-  DenseMap<StageSeparatingOp, llvm::MapVector<Value, Backedge>> stageRegMap;
+  DenseMap<Block *, llvm::MapVector<Value, Backedge>> stageRegMap;
 
   // A linked list of stages in the pipeline. Allows for easily looking up the
   // predecessor stage of a given stage.
-  DenseMap<StageSeparatingOp, StageSeparatingOp> stagePredecessor;
+  DenseMap<Block *, Block *> stagePredecessor;
```

Comment:
This wouldn't be necessary if you used the numerical order of the blocks...

Reply:
Oops, actually thought i'd purged all of those uses in favor of predecessor/successor blocks.